src/objective-c/GRPCClient/private/GRPCWrappedCall.m

```diff
@@ -0,0 +1,193 @@
+        opBlock = ^{
+          return @YES;
+        };
+        break;
+      case GRPC_OP_RECV_INITIAL_METADATA:
+        recv_initial_metadata = gpr_malloc(sizeof(grpc_metadata_array));
+        grpc_metadata_array_init(recv_initial_metadata);
+        current->data.recv_initial_metadata = recv_initial_metadata;
+        opBlock = ^{
+          NSDictionary *metadata = [NSDictionary grpc_dictionaryFromMetadata:recv_initial_metadata->metadata count:recv_initial_metadata->count];
+          grpc_metadata_array_destroy(recv_initial_metadata);
+          return metadata;
+        };
+        break;
+      case GRPC_OP_RECV_MESSAGE:
+        recv_message = gpr_malloc(sizeof(grpc_byte_buffer*));
+        current->data.recv_message = recv_message;
+        opBlock = ^{
+          NSData *data = [NSData grpc_dataWithByteBuffer:*recv_message];
+          grpc_byte_buffer_destroy(*recv_message);
+          gpr_free(recv_message);
+          return data;
+        };
+        break;
+      case GRPC_OP_RECV_STATUS_ON_CLIENT:
+        status_code = gpr_malloc(sizeof(status_code));
+        current->data.recv_status_on_client.status = status_code;
+        status_details = gpr_malloc(sizeof(char*));
+        *status_details = NULL;
+        current->data.recv_status_on_client.status_details = status_details;
+        status_details_capacity = gpr_malloc(sizeof(grpc_status_code));
+        *status_details_capacity = 0;
+        current->data.recv_status_on_client.status_details_capacity = status_details_capacity;
+        recv_trailing_metadata = gpr_malloc(sizeof(grpc_metadata_array));
+        grpc_metadata_array_init(recv_trailing_metadata);
+        current->data.recv_status_on_client.trailing_metadata = recv_trailing_metadata;
+        opBlock = ^{
+          grpc_status status;
+          status.status = *status_code;
+          status.details = *status_details;
+          status.metadata = recv_trailing_metadata;
+          gpr_free(status_code);
+          gpr_free(status_details);
+          gpr_free(status_details_capacity);
+          return [NSError grpc_errorFromStatus:&status];
+        };
+        break;
+      case GRPC_OP_SEND_STATUS_FROM_SERVER:
+        [NSException raise:NSInvalidArgumentException format:@"Not a server: cannot send status"];
+      default:
+        [NSException raise:NSInvalidArgumentException format:@"Unrecognized dictionary key"];
+    }
+    current->op = [key intValue];
+    [opProcessors setObject:opBlock forKey:key];
+  }
+  grpc_call_error error = grpc_call_start_batch(call, ops_array, nops, (__bridge_retained void *)(^(grpc_op_error error){
+    if (error != GRPC_OP_OK) {
+      [NSException raise:@"Operation Exception" format:@"The batch failed with an unknown error"];
+    }
+    NSMutableDictionary *result = [NSMutableDictionary new];
+    for (id key in opProcessors) {
+      id(^block)(void) = opProcessors[key];
+      id value = block();
+      if (value == nil) {
+        value = [NSNull null];
```

Comment:
Ah, here's the [NSNull null]! So it seems you can just do:

    if (value) {
      result[key] = value;
    }
?

But we're again reimplementing static typing by hand, via the use of type tags. So instead of having the block return anything, and forcing the writer of handleCompletion to give up typing, why not have opBlock _call_ a well-typed handler, provided by whoever constructed the Operation? For example, when the caller of startBatch:... constructs the Operation corresponding to the GRPC_OP_RECV_STATUS_ON_CLIENT tag, it can pass to its initializer the handler for the NSError*. Then when this operation object is asked to return an opProcessor, it'll return one that constructs the NSError and passes it to the handler.